### ISSUE TITLE
log autopicker fails with mpi

### ISSUE THREAD

**Issue by azazellochg** [scipion-em/scipion-em-relion#64]
Is it supposed to run with MPIs?
```
[35mSTARTED[0m: pickMicrographListStep, step 1
  2019-05-23 20:47:37.798279
Picking micrograph: Runs/007659_ProtRelionMotioncor/extra/20170629_00022_frameImage_aligned_mic_DW.mrc 
Picking micrograph: Runs/007659_ProtRelionMotioncor/extra/20170629_00023_frameImage_aligned_mic_DW.mrc 
Picking micrograph: Runs/007659_ProtRelionMotioncor/extra/20170629_00029_frameImage_aligned_mic_DW.mrc 
Picking micrograph: Runs/007659_ProtRelionMotioncor/extra/20170629_00047_frameImage_aligned_mic_DW.mrc 
Sending command to 2: cwd=Runs/008032_ProtRelionAutopickLoG
Sending environment to 2
Sending command to 2: relion_autopick_mpi  --pickname autopick --odir "" --LoG --shrink 0 --lowpass 20.000  --i input_micrographs_2-22.star --LoG_diam_min 150.000 --LoG_diam_max 180.000 --LoG_adjust_threshold 0.000
Traceback (most recent call last):
  File "/net/nfs1/public/EM/Scipion/scipion-2.0.0/pyworkflow/protocol/executor.py", line 151, in run
    self.step._run()  # not self.step.run() , to avoid race conditions
  File "/net/nfs1/public/EM/Scipion/scipion-2.0.0/pyworkflow/protocol/protocol.py", line 237, in _run
    resultFiles = self._runFunc()
  File "/net/nfs1/public/EM/Scipion/scipion-2.0.0/pyworkflow/protocol/protocol.py", line 233, in _runFunc
    return self._func(*self._args)
  File "/net/nfs1/public/EM/Scipion/scipion-2.0.0/pyworkflow/em/protocol/protocol_particles_picking.py", line 280, in pickMicrographListStep
    self._pickMicrographList(micList, *args)
  File "/net/nfs1/public/EM/Scipion/scipion-2.0.0/software/lib/python2.7/site-packages/relion/protocols/protocol_autopick_v3.py", line 169, in _pickMicrographList
    self._pickMicrographsFromStar(micStar, *args)
  File "/net/nfs1/public/EM/Scipion/scipion-2.0.0/software/lib/python2.7/site-packages/relion/protocols/protocol_autopick_v3.py", line 152, in _pickMicrographsFromStar
    self.runJob(program, params, cwd=self.getWorkingDir())
  File "/net/nfs1/public/EM/Scipion/scipion-2.0.0/pyworkflow/protocol/protocol.py", line 1311, in runJob
    self._stepsExecutor.runJob(self._log, program, arguments, **kwargs)
  File "/net/nfs1/public/EM/Scipion/scipion-2.0.0/pyworkflow/protocol/executor.py", line 353, in runJob
    gpuList=self.getGpuList())
  File "/net/nfs1/public/EM/Scipion/scipion-2.0.0/pyworkflow/utils/mpi.py", line 88, in runJobMPI
    send(command, mpiComm, mpiDest, TAG_RUN_JOB+mpiDest)
  File "/net/nfs1/public/EM/Scipion/scipion-2.0.0/pyworkflow/utils/mpi.py", line 73, in send
    raise Exception(str(result))
Exception: Command 'relion_autopick_mpi  --pickname autopick --odir "" --LoG --shrink 0 --lowpass 20.000  --i input_micrographs_2-22.star --LoG_diam_min 150.000 --LoG_diam_max 180.000 --LoG_adjust_threshold 0.000' returned non-zero exit status 1
[31mProtocol failed: Command 'relion_autopick_mpi  --pickname autopick --odir "" --LoG --shrink 0 --lowpass 20.000  --i input_micrographs_2-22.star --LoG_diam_min 150.000 --LoG_diam_max 180.000 --LoG_adjust_threshold 0.000' returned non-zero exit status 1[0m
[35mFAILED[0m: pickMicrographListStep, step 1
  2019-05-23 20:47:41.925975
[32m------------------- PROTOCOL FAILED (DONE 1/2)[0m
```

Also, threads option does not seem to do anything. There is no thread param for relion cmd apparently.

**Comment by azazellochg** [comment]
It's true that I ran this with batch=0 which is a bad idea to do with mpi. Re-run with batch=1, same error.

**Comment by delarosatrevin** [comment]
Batch 0 is fine, it will take all available micrographs and join in a batch...but it is true that with current PARALLE_STEPS mode might not be the best way to run it. I need to check MPI in multiple nodes, in a single workstations it seems to be working.

**Comment by delarosatrevin** [comment]
Ah, it is also failing locally in my pc with MPIs....I will check.

**Comment by delarosatrevin** [comment]
Fixed in 6f09e85
